packages/asset-service/src/service/AssetService.ts

```diff
@@ -8,8 +8,8 @@ export class AssetService {
   private assetData: BaseAsset[]
   private flatAssetData: Asset[]
 
-  constructor(assetFileUrl: string) {
-    this.assetFileUrl = assetFileUrl
+  constructor(assetFileUrl?: string) {
+    this.assetFileUrl = assetFileUrl || ''
```

Comment:
What happens if `assetFileUrl` is empty?

My preference would be to change the private variable type to `string | undefined` rather than filling it with an empty string. Then you can do a `if (typeof this.assetFileUrl === 'string)` check to see if it's been set.

Reply:
`assetFileUrl` isn't used in the new `description` function and you didn't change anywhere that uses this, so I'm unsure why this change is here.

Reply:
I wanted to make it optional. I agree with you on letting it be undefined rather than empty string

Reply:
i pushed up a fix. lmk what you think